From 8d3e537724a1b72865d22e0b013be84a6a36afef Mon Sep 17 00:00:00 2001
From: matsfunk <mathias.funk@gmail.com>
Date: Thu, 13 Jul 2023 21:43:46 +0200
Subject: [PATCH]  - some boilerplate clean-up  - added data-id attribute to
 root element to be able to inject references  - styling in App.css to allow
 for embedding Data Foundry view  - changed data source  - fixed imports and
 font size for drawing/styling nodes

---
 public/index.html    |  6 +++---
 public/manifest.json |  4 ++--
 src/App.css          | 11 ++++++-----
 src/App.js           |  3 ++-
 src/styleNode.js     |  7 +++++--
 5 files changed, 18 insertions(+), 13 deletions(-)

diff --git a/public/index.html b/public/index.html
index aa069f2..8125689 100644
--- a/public/index.html
+++ b/public/index.html
@@ -7,7 +7,7 @@
     <meta name="theme-color" content="#000000" />
     <meta
       name="description"
-      content="Web site created using create-react-app"
+      content=""
     />
     <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
     <!--
@@ -24,11 +24,11 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>React App</title>
+    <title>TwineJS Researcher Web</title>
   </head>
   <body>
     <noscript>You need to enable JavaScript to run this app.</noscript>
-    <div id="root"></div>
+    <div id="root" data-id=""></div>
     <!--
       This HTML file is a template.
       If you open it directly in the browser, you will see an empty page.
diff --git a/public/manifest.json b/public/manifest.json
index f19b3ec..0217cad 100644
--- a/public/manifest.json
+++ b/public/manifest.json
@@ -1,6 +1,6 @@
 {
-  "short_name": "React App",
-  "name": "Create React App Sample",
+  "short_name": "Researcher Web",
+  "name": "Narrative Surveys Researcher Web",
   "start_url": ".",
   "display": "standalone",
   "theme_color": "#000000",
diff --git a/src/App.css b/src/App.css
index 0970f4e..6f5f5cb 100644
--- a/src/App.css
+++ b/src/App.css
@@ -1,9 +1,10 @@
 .App {
-  position: fixed;
-  top: 0;
-  left: 0;
-  bottom: 0;
-  right: 0;
+  width: 100%;
+  overflow-y: auto;
+  height: calc(100vh - 250px);
+  position: relative;
+  background: white;
+  font-size: 0.8rem;
   overflow: auto;
 }
 
diff --git a/src/App.js b/src/App.js
index 50d44cb..c313370 100644
--- a/src/App.js
+++ b/src/App.js
@@ -5,7 +5,8 @@ import transformData from "./transformData";
 import renderForeignObjectNode from "./styleNode";
 
 function App() {
-  var data = require("./data.json");
+  var data = window.narrativeSurveyDataJson;
+  // var data = require("./data.json");
   const transformedData = transformData(data);
   const foreignObjectProps = { width: 200, height: 300, x: 50, y: -30 };
   return (
diff --git a/src/styleNode.js b/src/styleNode.js
index 20e171e..75c4a5c 100644
--- a/src/styleNode.js
+++ b/src/styleNode.js
@@ -1,3 +1,6 @@
+import React from "react";
+import ReactDOM from "react-dom/client";
+
 const renderForeignObjectNode = ({
   nodeDatum,
   toggleNode,
@@ -21,7 +24,7 @@ const renderForeignObjectNode = ({
         text-anchor="middle"
         alignment-baseline="central"
         fill="white"
-        font-size={dynamicSize}
+        font-size="1rem"
         stroke-width="0px"
       >
         {peopleCount}
@@ -31,7 +34,7 @@ const renderForeignObjectNode = ({
         <div
           style={{ display: "flex", flexDirection: "column", height: "100%" }}
         >
-          <h3 style={{ fontWeight: 300 }}>{nodeDatum.name}</h3>
+          <span class="nodeText" style={{ fontWeight: 300 }}>{nodeDatum.name}</span>
           {isInput && (
             <div
               style={{
-- 
2.39.1

From ebed713137ee653e0e510277a630ca285e0b69d6 Mon Sep 17 00:00:00 2001
From: matsfunk <mathias.funk@gmail.com>
Date: Thu, 13 Jul 2023 21:44:01 +0200
Subject: [PATCH]  - changed dependency

---
 package.json | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package.json b/package.json
index eead421..d5692cc 100644
--- a/package.json
+++ b/package.json
@@ -6,7 +6,7 @@
     "react": "^18.2.0",
     "react-d3-tree": "^3.5.2",
     "react-dom": "^18.2.0",
-    "react-scripts": "5.0.1"
+    "react-scripts": "^2.1.3"
   },
   "scripts": {
     "start": "react-scripts start",
-- 
2.39.1

